doc/source/security/data-masking.rst

```diff
@@ -0,0 +1,307 @@
+
+              +--------------------------------------+
+              | gen_range(10,100)                    |
+              +--------------------------------------+
+              | 56                                   |
+              +--------------------------------------+
+
+              mysql> SELECT gen_range(-100,-80);
+
+              +--------------------------------------+
+              | gen_range(-100,-80)                    |
+              +--------------------------------------+
+              | -91                             |
+              +--------------------------------------+
+
+    * - gen_rnd_email()
+      - Generates a random email address. The domain is ``example.com``.
+
+      - .. code-block:: MySQL
+
+             mysql> SELECT gen_rnd_email();
+
+             +---------------------------------------+
+             | gen_rnd_email()                       |
+             +---------------------------------------+
+             | sma.jrts@example.com                  |
+             +---------------------------------------+
+
+    * - gen_rnd_pan([size in integer])
+      - Generates a random primary account number. This function should only
+        be used for test purposes.
+
+      - .. code-block:: MySQL
+
+              mysql> SELECT mask_pan(gen_rnd_pan());
+
+              +-------------------------------------+
+              | mask_pan(gen_rnd_pan())             |
+              +-------------------------------------+
+              | XXXXXXXXXXXX4444                    |
+              +-------------------------------------+
+
+    * - gen_rnd_us_phone()
+      - Generates a random U.S. phone number. The generated number adds the
+        `1` dialing code and is in the `555` area code. The `555` area code
+        is not valid for any U.S. phone number.
+      - .. code-block:: MySQL
+
+            mysql> SELECT gen_rnd_us_phone();
+
+            +-------------------------------+
+            | gen_rnd_us_phone()            |
+            +-------------------------------+
+            | 1-555635-5709                 |
+            +-------------------------------+
+
+    * - gen_rnd_ssn()
+      - Generates a random, non-legitimate US Social Security Number in
+        an ``AAA-BBB-CCCC`` format. This function should only be used for test
+        purposes.
+      - .. code-block:: MySQL
+
+          mysql> SELECT gen_rnd_ssn()
+
+          +-----------------------------+
+          | gen_rnd_ssn()               |
+          +-----------------------------+
+          | 995-33-5656                 |
+          +-----------------------------+
+
+.. rubric:: Using Dictionaries to Generate Random Terms
+
+Use a selected dictionary to generate random terms. The dictionary must
+be loaded from a file with the following characteristics:
+
+* Plain text
+
+* One term per line
+
+* Must contain at least one entry
+
+The following table displays the commands for using dictionaries to generate random terms:
+
```

Comment:
better to show example files with contents

Reply:
Added example